### PR TITLE
tests plus working on adding next enemy

### PR DESCRIPTION
**Pull request by snowyboyo**
I wrote three tests, so far all having to do with collision because it made the most sense to me to start with. also started working adding the next enemy and practiced touch typing

### CODE REVIEW

File: GunShipAlpha/src/Behemoth.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import java.awt.*;
+import java.awt.geom.Point2D;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
```
Comment (jlb0170) [inline GunShipAlpha/src/Behemoth.java:5]: nope, tank and behemoth are identical, or nearly so. you should have guessed that i'd immediately say no to that. 

if you need a pointer on how to fix it, look at "inheritance". There's a few other ways, as well, but inheritance from a common ancestor is probably the easiest.

Comment (jlb0170) [reply GunShipAlpha/src/Behemoth.java:5]: also, please don't submit pull requests with duplication like this - as i said, you should _know_ instinctively that this is wrong. if you don't know how to fix it, throw up a hand and ask what to do, but it makes me feel like I haven't taught you the most basic thing about duplication when you submit a PR like this :)

File: GunShipAlpha/src/Game.java [hunk 2]
```diff
@@ -95,8 +96,18 @@ public void spawnBehemoths(int count) {
...
             Game.spawnEnemiesRandomLocation result = getSpawnEnemiesRandomLocation(side, y, x, rand);
-            Enemy enemy = new Enemy(result.x(), result.y(), getWidth() / 2, getHeight() / 2);
-            gs.addEnemy(enemy);
+            Behemoth behemoth = new Behemoth(result.x(), result.y(), getWidth() / 2, getHeight() / 2);
+            gs.addBehemoth(behemoth);
+        }
+    }
+    public void spawnTanks(int count) {
+        Random rand = new Random();
+        for (int i = 0; i < count; i++) {
```
Comment (jlb0170) [inline GunShipAlpha/src/Game.java:104]: it seems like this and spawnBehemoths both need to check to ensure that the location doesn't put the next one on top of any of the existing ones...

File: GunShipAlpha/src/GameState.java [hunk 3]
```diff
@@ -6,39 +6,58 @@
...
+            if (!isTankBlocked(tank)) {
+                tank.moveTowardTarget();
+                removeEnemy();
+            }
+        }
+    }
 
 
-    boolean isEnemyBlocked(Enemy enemy) {
+    boolean isBehemothBlocked(Behemoth enemy) {
```
Comment (jlb0170) [inline GunShipAlpha/src/GameState.java:37]: removing the tank/behemoth duplication will also allow you to remove this as well. I want you to see this kind of stuff and recognize it as wrong, like scattering tacks on your bed. duplication is bad.

File: GunShipAlpha/src/Tests.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Test
+    void testBehemothPlayerCollision() {
+        GameState gameState = new GameState();
+        Player player = new Player();
+        Behemoth behemoth = new Behemoth(350, 350, 50, 50);
+        gameState.addBehemoth(behemoth);
+        behemoth.moveTowardTarget();
+        boolean isCollision = gameState.checkBehemothPlayerCollisions(player, 800, 600);
+        assertTrue(isCollision);
```
Comment (jlb0170) [inline GunShipAlpha/src/Tests.java:19]: i don't see any tests that assertFalse collision - seems like you're missing some cases. 

Also, with all these hard coded values, you should at least put in a comment explaining what the arrangement represents or something.

**Review by jlb0170 — CHANGES_REQUESTED** [review]

File: GunShipAlpha/src/Behemoth.java (hunk 1 above)
Comment (snowyboyo) [reply GunShipAlpha/src/Behemoth.java:5]: that's fair, I'll fix it

File: GunShipAlpha/src/Game.java (hunk 2 above)
Comment (snowyboyo) [reply GunShipAlpha/src/Game.java:104]: fixed

File: GunShipAlpha/src/GameState.java (hunk 3 above)
Comment (snowyboyo) [reply GunShipAlpha/src/GameState.java:37]: fixed

File: GunShipAlpha/src/Tests.java (hunk 4 above)
Comment (snowyboyo) [reply GunShipAlpha/src/Tests.java:19]: wdym by assert false collisions? like test a scenario that shouldn't be a collision?

Comment (jlb0170) [reply GunShipAlpha/src/Tests.java:19]: yes, exactly. different tests should (try to) cover every path the code can take, not just the "happy path" or most obvious case.

**Pull request merged**